agent/main.go

```diff
@@ -349,3 +388,199 @@ func mainImpl() {
 		os.Exit(1)
 	}
 }
+
+// Watch for CM creation.
+func watchConfigMaps(cfg *agentConfig) {
+	source := cache.NewListWatchFromClient(
+		cfg.KubeClient.Core().RESTClient(),
+		"configmaps",
+		"weave",
+		fields.SelectorFromSet(fields.Set{"metadata.name": "cloudwatch"}))
+
+	cfg.CMInformer = cache.NewSharedIndexInformer(
+		source,
+		&apiv1.ConfigMap{},
+		0,
+		cache.Indexers{},
+	)
+
+	cfg.CMInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    cfg.handleCMAdd,
+		UpdateFunc: cfg.handleCMUpdate,
+		DeleteFunc: cfg.handleCMDelete,
+	})
+}
+
+// Triggered on all ConfigMap creation with name cloudwatch in weave ns
+func (cfg *agentConfig) handleCMAdd(obj interface{}) {
+	cm, ok := obj.(*apiv1.ConfigMap)
+	if !ok {
+		log.Error("Failed to type assert ConfigMap")
```

Comment:
Nit: `!ok` seems unlikely to happen, but if it does, then would it be worth logging the type of the object?